python/cudf/cudf/core/dataframe.py

```diff
@@ -1565,6 +1577,9 @@ def __deepcopy__(self, memo={}):
             memo = {}
         return self.copy(deep=True)
 
+    def __reduce__(self):
+        return (DataFrame, (self._data, self.index))
```

Comment:
Can this not be handled at the Table / Frame level by using `self.__class__`?

Reply:
This would require all `Frame` subclasses to be constructible from an `OrderedColumnDict` and an `Index`, and that might not always make sense (e.g., you can't pass `index=` to an `Index` constructor). We could get around this with `**kwargs` in the constructor but I think that's icky too.